### PR TITLE
Feature/improve ternary operator evaluation

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Improve lazy evaluation capabilities of ternary operator.
Work in progress. Not relevant for 3.6.0.

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)

### CODE REVIEW

**Comment by jsteemann** [comment]
@gnusi: do you think it is a problem for arangosearch if we introduce AstNodes of type `NODE_TYPE_OPERATOR_TERNARY` with just two members (the condition and the false part)?

I checked the code and found that arangosearch uses `arangodb::aql::Expression::execute()` to execute the ternary operator. So to me it looks like no extra change is required for arangosearch to support this.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8364/

**Comment by gnusi** [comment]
@jsteeman at glance I don't see any problems

**Comment by jsteemann** [comment]
Tests blue

File: tests/js/server/aql/aql-ternary.js [hunk 1]
```diff
@@ -246,7 +255,7 @@ function ahuacatlTernaryCollectionTestSuite () {
       db._drop(cn);
       c = db._create(cn);
       for (var i = 0; i < 100; ++i) {
-        c.insert({value:i});
+        c.insert({_key: "test" + i, value: i});
```
Comment (dothebart) [inline tests/js/server/aql/aql-ternary.js]: please make this a bulk insert

Comment (jsteemann) [reply tests/js/server/aql/aql-ternary.js]: It just 100 documents inserted in a single test case... Anyway, I did it.

**Pull request merged**